cli/pkg/helm/helm_helper.go

```diff
@@ -178,21 +178,35 @@ func (c Helper) UpgradeChart(ch *chart.Chart, releaseName, namespace string, val
 		histClient := action.NewHistory(cfg)
 		var release *release.Release
 
-		if _, err = histClient.Run(releaseName); err == driver.ErrReleaseNotFound {
+		timeoutInMinutes := 10
+
+		if releases, err := histClient.Run(releaseName); err == driver.ErrReleaseNotFound {
+			// fresh install
 			iCli := action.NewInstall(cfg)
 			iCli.Namespace = namespace
 			iCli.ReleaseName = releaseName
 			iCli.Wait = true
-			iCli.Timeout = 10 * time.Minute
+			iCli.Timeout = time.Duration(timeoutInMinutes) * time.Minute
 			release, err = iCli.Run(ch, vals)
 		} else {
+			logging.PrintLog("Found existing installation, overwriting...", logging.InfoLevel)
+
+			// check if the previous installation is still pending (e.g., waiting to complete)
+			for _, release := range releases {
+				if release.Info.Status == "pending-upgrade" {
```

Comment:
should we also check for `release.StatusPendingInstall, release.StatusPendingUpgrade and release.StatusPendingRollback` here?

Reply:
Good idea. As we were already using `release` as a variable, I renamed it to `newRelease` and changed the if clause.